Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -5176,13 +5176,22 @@ void WebPageProxy::runJavaScriptInFrameInScriptWorld(RunJavaScriptParameters&& p
         activity = m_process->throttler().foregroundActivity("WebPageProxy::runJavaScriptInFrameInScriptWorld"_s);
 #endif
 
-    sendWithAsyncReply(Messages::WebPage::RunJavaScriptInFrameInScriptWorld(parameters, frameID, world.worldData()), [activity = WTFMove(activity), callbackFunction = WTFMove(callbackFunction)] (const IPC::DataReference& dataReference, std::optional<ExceptionDetails>&& details) mutable {
+    auto completionHandler = [activity = WTFMove(activity), callbackFunction = WTFMove(callbackFunction)] (const IPC::DataReference& dataReference, std::optional<ExceptionDetails>&& details) mutable {
         if (details)
             return callbackFunction(makeUnexpected(WTFMove(*details)));
         if (dataReference.empty())
             return callbackFunction({ nullptr });
         callbackFunction({ API::SerializedScriptValue::createFromWireBytes(Vector(dataReference)).ptr() });
-    });
+    };
+
+    if (auto* frame = frameID ? WebFrameProxy::webFrame(*frameID) : nullptr) {
+        if (auto remotePageProxy = frame->remotePageProxy()) {
+            remotePageProxy->sendWithAsyncReply(Messages::WebPage::RunJavaScriptInFrameInScriptWorld(parameters, frameID, world.worldData()), completionHandler);
```

Comment:
WTFMove(completionHandler) to avoid unnecessary copies.  Same below.

Also, fun fact: you can return the result of sendWithAsyncReply and then you don't need the { } because it's one line.  Some people don't like that style, but I do.  Ok either way.